test/firebase_storage_mocks_test.dart

```diff
@@ -115,9 +118,46 @@ void main() {
         expect(downloadUrl.contains('some-bucket/o/someimage.png'), isTrue);
       });
     });
+
+    test('Ref listAll', () async {
+      final basePath = 'this/is/basic';
+      final otherPath = 'another/path';
+      final storage = MockFirebaseStorage();
+      await storage.ref(basePath + '/data2').putData(randomData(255));
+      await storage.ref(basePath + '/subdir1/data1').putData(randomData(255));
+      await storage.ref(basePath + '/data2').putData(randomData(255));
+      await storage.ref(otherPath + '/data3').putData(randomData(255));
```

Comment:
It'd have been nice to use vary the file names a little to make the tests easier to read. We do expect a file called data3, but it's the one from basePath.